x-pack/legacy/plugins/siem/server/kibana.index.ts

```diff
@@ -27,33 +25,29 @@ import { ServerFacade } from './types';
 const APP_ID = 'siem';
 
 export const initServerWithKibana = (
-  kbnServer: ServerFacade,
-  logger: Logger,
-  mode: EnvironmentMode
+  context: PluginInitializerContext,
+  { plugins: { alerting, xpack_main }, route }: ServerFacade
 ) => {
-  if (kbnServer.plugins.alerting != null) {
-    const version = kbnServer.config().get<string>('pkg.version');
+  const logger = context.logger.get('plugins', APP_ID);
+  const version = context.env.packageInfo.version;
+
+  if (alerting != null) {
     const type = rulesAlertType({ logger, version });
     if (isAlertExecutor(type)) {
-      kbnServer.plugins.alerting.setup.registerType(type);
+      alerting.setup.registerType(type);
     }
   }
-  kbnServer.injectUiAppVars('siem', async () => kbnServer.getInjectedUiAppVars('kibana'));
-
-  const libs = compose(kbnServer, mode);
-  initServer(libs);
 
   // Signals/Alerting Rules routes for
   // routes such as ${DETECTION_ENGINE_RULES_URL}
   // that have the REST endpoints of /api/detection_engine/rules
-  createRulesRoute(kbnServer);
-  readRulesRoute(kbnServer);
-  updateRulesRoute(kbnServer);
-  deleteRulesRoute(kbnServer);
-  findRulesRoute(kbnServer);
+  createRulesRoute(route);
+  readRulesRoute(route);
+  updateRulesRoute(route);
+  deleteRulesRoute(route);
+  findRulesRoute(route);
 
-  const xpackMainPlugin = kbnServer.plugins.xpack_main;
-  xpackMainPlugin.registerFeature({
+  xpack_main.registerFeature({
```

Comment:
Should I move this into the NP plugin, or wait until the NP equivalent is available to do so?

Reply:
You can use the NP `features` plugin: https://github.com/elastic/kibana/blob/99c6396fd571074f8bc09e723ac44009c114c01b/x-pack/plugins/features/server/plugin.ts#L27

Reply:
c7bf344